Makefile

```diff
@@ -1,111 +1,115 @@
-GCR_URL = us.gcr.io/vcm-ml
-DOCKERFILE ?= docker/Dockerfile
-COMPILED_TAG_NAME ?= latest
-ENVIRONMENT_TAG_NAME ?= latest
-COMPILE_OPTION ?=
-COMPILE_TARGET ?= fv3gfs-compiled
-BUILD_ARGS ?=
+# setup (use XXX=<value> make <target> to override)
+GCR_URL                 ?= us.gcr.io/vcm-ml
+DOCKERFILE              ?= docker/Dockerfile
+COMPILED_TAG_NAME       ?= latest
+ENVIRONMENT_TAG_NAME    ?= latest
+COMPILE_OPTION          ?=
+COMPILE_TARGET          ?= fv3gfs-compiled
+BUILD_ARGS              ?=
 BUILD_FROM_INTERMEDIATE ?= n
-ENVIRONMENT_TARGET ?= fv3gfs-environment
-COMPILED_IMAGE ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)
-SERIALIZE_IMAGE ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)-serialize
-ENVIRONMENT_IMAGE=$(GCR_URL)/$(ENVIRONMENT_TARGET):$(ENVIRONMENT_TAG_NAME)
-IMAGE ?= $(ENVIRONMENT_IMAGE)
-
-FMS_IMAGE = $(GCR_URL)/fms-build
-ESMF_IMAGE = $(GCR_URL)/esmf-build
-SERIALBOX_IMAGE = $(GCR_URL)/serialbox-build
-
-MOUNTS?=-v $(shell pwd)/FV3:/FV3 \
-	-v $(shell pwd)/FV3/conf/configure.fv3.gnu_docker:/FV3/conf/configure.fv3
-
-MOUNTS_SERIALIZE?=-v $(shell pwd)/FV3:/FV3/original
-
-EXPERIMENT ?= new
-RUNDIR_CONTAINER=/rundir
-RUNDIR_HOST=$(shell pwd)/rundir
+ENVIRONMENT_TARGET      ?= fv3gfs-environment
 
+# image names (use XXX_IMAGE=<name> make <target> to override)
+COMPILED_IMAGE     ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)
+SERIALIZE_IMAGE    ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)-serialize
+ENVIRONMENT_IMAGE  ?= $(GCR_URL)/$(ENVIRONMENT_TARGET):$(ENVIRONMENT_TAG_NAME)
+FMS_IMAGE          ?= $(GCR_URL)/fms-build
+ESMF_IMAGE         ?= $(GCR_URL)/esmf-build
+SERIALBOX_IMAGE    ?= $(GCR_URL)/serialbox-build
 
+# used to shorten build times in CircleCI
 ifeq ($(BUILD_FROM_INTERMEDIATE),y)
 	BUILD_ARGS += --build-arg FMS_IMAGE=$(FMS_IMAGE) --build-arg ESMF_IMAGE=$(ESMF_IMAGE) --build-arg SERIALBOX_IMAGE=$(SERIALBOX_IMAGE)
 endif
 
-build: build_compiled
+.PHONY: help
+help:
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
-build_environment:
+.PHONY: build
+build: build_compiled  ## build default container image (production)
+
+.PHONY: build_environment
+build_environment:  ## build environment container image
 	docker build -f $(DOCKERFILE) -t $(ENVIRONMENT_IMAGE) \
 	--target $(ENVIRONMENT_TARGET) .
 
-build_compiled:
+.PHONY: build_compiled
+build_compiled:  ## build production container image
 	docker build \
 		--build-arg compile_option=$(COMPILE_OPTION) \
 		$(BUILD_ARGS) \
 		-f $(DOCKERFILE) \
 		-t $(COMPILED_IMAGE) \
 		--target $(COMPILE_TARGET) .
 
-build_serialize:
-	BUILD_ARGS="$(BUILD_ARGS) --build-arg serialize=true" COMPILED_IMAGE=$(SERIALIZE_IMAGE) $(MAKE) build_compiled
+.PHONY: build_serialize_check_md5
```

Comment:
I would prefer to keep the convention and update them as noticed/needed. As long as we aren't actually using the make targets as filenames/directory names, it is not necessary for .PHONY to be up to date. It is a little jarring to see an extra copy of the target with every target in the makefile.

We could discuss this as a group if you think this would be a significant improvement.

Reply:
This is the third time I work on this Makefile, everytime the PHONY's were not updated by anyone. I've moved them back in grouped format to the front of the file, in the hope that this will make them more visible.

Reply:
Thanks for the compromise!